youtube_dl/extractor/newgrounds.py

```diff
@@ -59,47 +61,64 @@ def _real_extract(self, url):
         webpage = self._download_webpage(url, media_id)
 
         title = self._html_search_regex(
-            r'<title>([^>]+)</title>', webpage, 'title')
+            r'<title>(.+?)</title>', webpage, 'title')
+
+        try:
+            media_url = self._parse_json(self._search_regex(
+                r'"url"\s*:\s*("[^"]+"),', webpage, ''), media_id)
+        except RegexNotFoundError or ExtractorError:
+            media_url = None
+        formats = []
+
+        if media_url:
+            formats = [{
+                'url': media_url,
+                'format_id': 'source',
+                'quality': 1,
+            }]
+        else:
+            json_data = self._download_json('https://www.newgrounds.com/portal/video/' + media_id, media_id, headers={
+                'Accept': 'application/json, text/javascript, */*; q=0.01',
+                'Accept-Encoding': 'gzip, deflate, br',
+                'X-Requested-With': 'XMLHttpRequest',
+                'Connection': 'keep-alive',
+            })
+
+            for resolution in ('360p', '720p', '1080p'):
```

Comment:
If they send (eg) 480p, we won't find it; also we don't know that the URL is there. How about s/t like:
```
            for resolution in filter(lambda x: re.match(r'\d+p', x), json_data['sources'].keys()):
                url = try_get(json_data['sources'][resolution], lambda x: x[0]['src'], compat_str)
                if not url:
                    continue
                formats.append({
                    'url': url,
                    ...
```
Or perhaps try anything that parses as a resolution:
```
for resolution, parsed_resolution in filter(lambda x: x[1], map(lambda x: (x, parse_resolution(x),), json_data['sources'].keys())):
                url = try_get(json_data['sources'][resolution], lambda x: x[0]['src'], compat_str)
                if not url:
                    continue
                formats.append({
                    'url': url,
                    'height': parsed_resolution.get('height'),
                    ...
```